src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -2209,35 +2207,81 @@ void PSParallelCompact::marking_phase(ParCompactionManager* cm,
   _gc_tracer.report_object_count_after_gc(is_alive_closure());
 }
 
-void PSParallelCompact::adjust_roots(ParCompactionManager* cm) {
-  // Adjust the pointers to reflect the new locations
-  GCTraceTime(Info, gc, phases) tm("Adjust Roots", &_gc_timer);
+class PSAdjustTask final : public AbstractGangTask {
+  SubTasksDone                               _sub_tasks;
+  WeakProcessor::Task                        _weak_proc_task;
+  OopStorageSetStrongParState<false, false>  _oop_storage_iter;
+  uint                                       _nworkers;
 
-  // Need new claim bits when tracing through and adjusting pointers.
-  ClassLoaderDataGraph::clear_claimed_marks();
+  enum PSAdjustSubTask {
+    PSAdjustSubTask_code_cache,
+    PSAdjustSubTask_aot,
+    PSAdjustSubTask_old_ref_process,
+    PSAdjustSubTask_young_ref_process,
```

Comment:
maybe use enum class, that way you wont have to use `PSAdjustSubTask` in the element names

Reply:
`SubTasksDone` APIs expect a `uint`, so I don't think `enum class` works here.